### ISSUE TITLE
Consider adding test-getDependencies.R to inst/tests/

### ISSUE THREAD

**Issue by danielturek** [nimble-dev/nimble#186]
Inspired by #185.

**Comment by paciorek** [comment]
We could create a basic test using the example in your email. Does that sound good?

**Comment by danielturek** [comment]
Sounds great.  If you're up for it, also perhaps including some usages of the various options arguments: self, stochOnly, determOnly, includeData, etc.

**Comment by perrydv** [comment]
Done.  Note that the options suggested in the previous comment are relevant for getNodeNames, not getDependencies.  We could extend test-getDependencies to a broader test-graph suite, but for now I kept it somewhat narrow on testing getDependencies.  

Along the way I realized that `test_package` use grepl to match names, so `test_package('nimble','mcmc')` will run *any* test-xyz.R file where xyz contains 'mcmc'.  This means that `test_package('nimble','mcmc')` was triggering test-mcmc, test-mcmc2, and test-mcmc3, which was not intended.  So I renamed the latter two test-Mcmc2 and test-Mcmc3 and used a similar scheme for test-Math2 and test-Filtering2.  We may want to discuss a better refactoring of large test topics later.

It may take a few more git steps to get this right since the GitHub desktop struggles with new and removed files.
